solver/src/encoding.rs

```diff
@@ -81,4 +128,33 @@ mod tests {
         assert_eq!(encoded[140..172], u8_as_32_bytes_be(10));
         assert_eq!(encoded[172..204], u8_as_32_bytes_be(11));
     }
+
+    #[test]
+    fn encode_uniswap_call_() {
```

Comment:
I always like unit tests, but here I'm not sure if we are not actually more testing ethcontracts rather than out code.

Reply:
Yeah I agree. Probably won't have a test like this for all the interactions. I was checking manually whether the bytes make sense and decided to keep the test.

Reply:
I'm fine with this test, as if we decide to not use `ethcontract` in the `encode_uniswap_call` implementation, we have a unit test that ensures it works :smile:.